app/storage/storage.py

```diff
@@ -0,0 +1,89 @@
+from abc import abstractmethod, ABC
+from functools import cached_property
+
+from flask import current_app
+
+from app.data_model import app_models
+
+
+class StorageModel:
+    TABLE_CONFIG = {
+        app_models.SubmittedResponse: {
+            "key_field": "tx_id",
+            "expiry_field": "expires_at",
+            "table_name_key": "EQ_SUBMITTED_RESPONSES_TABLE_NAME",
+            "schema": app_models.SubmittedResponseSchema,
+        },
+        app_models.QuestionnaireState: {
+            "key_field": "user_id",
+            "table_name_key": "EQ_QUESTIONNAIRE_STATE_TABLE_NAME",
+            "schema": app_models.QuestionnaireStateSchema,
+        },
+        app_models.EQSession: {
+            "key_field": "eq_session_id",
+            "expiry_field": "expires_at",
+            "table_name_key": "EQ_SESSION_TABLE_NAME",
+            "schema": app_models.EQSessionSchema,
+        },
+        app_models.UsedJtiClaim: {
+            "key_field": "jti_claim",
+            "expiry_field": "expires_at",
+            "table_name_key": "EQ_USED_JTI_CLAIM_TABLE_NAME",
+            "schema": app_models.UsedJtiClaimSchema,
+        },
+    }
+
+    def __init__(self, model=None, model_type=None):
+        self._model = model
+        self._model_type = type(model) if model else model_type
+
+        if not self._model_type:
+            raise ValueError("One of model/model_type is required")
```

Comment:
This could be simpler by making `model_type` required and move the responsibility of getting the type to the caller.

Reply:
I did this at first, then changed it so it doesn't have to provide both. But make sense to make it the responsibility of the caller therefore, have changed.

Reply:
I think we should drop model from the init, and pass it in the `serialize` call.

Reply:
Have dropped model, but re-introduces very small duplication in the storage classes, have a look, should be fine.